pay-api/src/pay_api/services/direct_pay_service.py

```diff
@@ -335,6 +336,7 @@ def get_receipt(
                 )
         return None
 
+    @cache.memoize(timeout=1800)
```

Comment:
This should work outside flask context, should speed stuff up too. The token was 30m on prod.